llpc/util/llpcBuilderDebug.h

```diff
@@ -0,0 +1,46 @@
+/*
+ ***********************************************************************************************************************
+ *
+ *  Copyright (c) 2020 Advanced Micro Devices, Inc. All Rights Reserved.
+ *
+ *  Permission is hereby granted, free of charge, to any person obtaining a copy
+ *  of this software and associated documentation files (the "Software"), to deal
+ *  in the Software without restriction, including without limitation the rights
+ *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ *  copies of the Software, and to permit persons to whom the Software is
+ *  furnished to do so, subject to the following conditions:
+ *
+ *  The above copyright notice and this permission notice shall be included in all
+ *  copies or substantial portions of the Software.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ *  SOFTWARE.
+ *
+ **********************************************************************************************************************/
+/**
+ ***********************************************************************************************************************
+ * @file  llpcBuilderDebug.h
+ * @brief LLPC header file: declaration of LLPC_OUTS for use in the middle-end
+ ***********************************************************************************************************************
+ */
+#pragma once
+
+#include "llvm/Support/raw_ostream.h"
+
+namespace Llpc
+{
+
+using namespace llvm;
+
+// Get pointer to stream for LLPC_OUTS, or nullptr if disabled.
+raw_ostream* GetLlpcOuts();
+
+} // Llpc
+
+// Output general message
+#define LLPC_OUTS(_msg) if (auto pStream = GetLlpcOuts()) { *pStream << _msg; }
```

Comment:
This may change code logic for the next `else` statement.

Can we do something like:
```
do { if (auto* pStream = GetLlpcOuts()) { *pStream << _msg; } } while (false)
```
instead?

Reply:
Also, `_msg` can be defined by system headers included above this one, no? Would't it be safer to have a name without a leading underscore?